### ISSUE TITLE
trivy does not exit with error in case of error

### ISSUE THREAD

**Issue by victornoel** [aquasecurity/trivy#3538]
## Description

I run trivy to generate a SBOM with:
```
trivy image --format cyclonedx --output sbom.json "myimage:mytag"
```

And I get this output:
```
2023-02-01T14:17:40.386Z	INFO	"--format cyclonedx" disables security scanning. Specify "--scanners vuln" explicitly if you want to include vulnerabilities in the CycloneDX report.
2023-02-01T14:17:48.139Z	INFO	JAR files found
2023-02-01T14:17:48.139Z	ERROR	Unable to initialize the Java DB: Java DB update failed: not initialized
```

## What did you expect to happen?

That the exit code is 1 so that my CI would fail.

## What happened instead?

The exit code was 0 and the CI didn't fail

## Output of run with `-debug`:

```
(paste your output here)
```

## Output of `trivy -v`:

```
trivy --version
Version: 0.37.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-02-01 00:12:43.097002183 +0000 UTC
  NextUpdate: 2023-02-01 06:12:43.097001783 +0000 UTC
  DownloadedAt: 2023-02-01 01:03:17.913006345 +0000 UTC
```

**Comment by knqyf263** [comment]
Thanks. I'm on it.

**Comment by knqyf263** [comment]
We're sorry for the inconvenience.

**Comment by knqyf263** [comment]
v0.37.1 should fix this issue.
https://github.com/aquasecurity/trivy/releases/tag/v0.37.1

**Comment by gpanagiotidis** [comment]
@knqyf263 I have a new issue now. The tag v0.37.1 causes scanning to be 5 times slower, and also receiving this error:

`Unable to initialize the Java DB: Java DB open error: can't open db: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub`

**Comment by victornoel** [comment]
> The tag v0.37.1 causes scanning to be 5 times slower

see https://github.com/aquasecurity/trivy-java-db/issues/8

> Unable to initialize the Java DB: Java DB open error: can't open db: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub

This is very suspicious because this is the main thing fixed by 0.37.1, see https://github.com/aquasecurity/trivy/releases/tag/v0.37.1

**Comment by gpanagiotidis** [comment]
> > The tag v0.37.1 causes scanning to be 5 times slower
> 
> see [aquasecurity/trivy-java-db#8](https://github.com/aquasecurity/trivy-java-db/issues/8)
> 
> > Unable to initialize the Java DB: Java DB open error: can't open db: Binary was compiled with 'CGO_ENABLED=0', go-sqlite3 requires cgo to work. This is a stub
> 
> This is very suspicious because this is the main thing fixed by 0.37.1, see https://github.com/aquasecurity/trivy/releases/tag/v0.37.1

You are right. When I reverted to 0.37.0 it appeared again.
